### PR TITLE
feat!: migrate to postgresql

### LINKED ISSUE #151: GCP subnet is created with project name instead of subnet name passed from the UI

### LINKED ISSUE #160: GCP convert for single vm is triggering convert for all instead of one

### LINKED ISSUE #221: Migration to postgres from cassandra

### PR DESCRIPTION
**Pull request by WeepingClown13**
Migration of cassandra db to postgres and related cleanup
closes #221 
This probably fixes #151, and fixes #160 as well.

ERD for database: 
postgres-migration-erd

### CODE REVIEW

File: model/blueprint.py [hunk 1]
```diff
@@ -1,59 +1,30 @@
...
+    status = Column(String)
+    image_id = Column(String)
+    vpc_id = Column(String)
+    subnet_id = Column(String)
+    public_route = Column(Boolean)
+    ig_id = Column(String)
+    route_table = Column(String)
+    vm_id = Column(String)
+    nic_id = Column(String)
+    disk_clone = Column(ARRAY(JSON(String)))
```
Comment (iamvishnuks) [inline model/blueprint.py:30]: disk_clone status could be moved to disk_status table with a foreign key relation to the id of this blueprint table

Comment (iamvishnuks) [reply model/blueprint.py:30]: I think we could use disk table here.

File: model/disk.py [hunk 2]
```diff
@@ -1,25 +1,14 @@
...
+class Disk(Base):
+    
+    __tablename__ = 'disk'
+    
+    project = Column(String, ForeignKey("project.name", onupdate='CASCADE', ondelete='CASCADE'), primary_key=True, unique=True)
+    host = Column(String, primary_key=True)
+    vhd = Column(String, nullable=False)
+    file_size = Column(String, nullable=False)
+    mnt_path = Column(String, nullable=False)
+    disk_id = Column(String, nullable=False)
```
Comment (iamvishnuks) [inline model/disk.py:14]: could include the disk clone status and blueprint_id here

File: model/network.py [hunk 3]
```diff
@@ -1,44 +1,24 @@
...
-        'indexes': [
-            {'fields': ('cidr', 'project','subnet_name'), 'unique': True}
-        ]
-    }
\ No newline at end of file
+class Subnet(Base):
+    
+    __tablename__ = 'subnet'
+    
+    project = Column(String, ForeignKey("project.name", onupdate='CASCADE', ondelete='CASCADE'), primary_key=True, unique=True)
```
Comment (iamvishnuks) [inline model/network.py:19]: there should be foreign key primary key relation between network table and subnet table

Comment (WeepingClown13) [reply model/network.py:19]: Network name is already a foreign key in Subnet.

**Comment by NihilBabu** [comment]
@WeepingClown13  clean up .env file and put placeholder for required values.

**Comment by WeepingClown13** [comment]
> @WeepingClown13  clean up .env file and put placeholder for required values.

There is no .env file. The env vaiables are read directly from the shell environment.

File: model/discover.py [hunk 4]
```diff
@@ -1,37 +1,24 @@
...
-    }
-
+    id = Column(String(40), primary_key=True)
+    created_at = Column(DateTime(timezone=True), nullable=False)
+    updated_at = Column(DateTime(timezone=True), nullable=False)
+    is_deleted = Column(Boolean, nullable=False, default=False)
+    is_locked = Column(Boolean, nullable=False, default=False)
+    is_enabled = Column(Boolean, nullable=False, default=True)
+    project = Column(String(40), ForeignKey("project.id"), nullable=False)
+    hostname = Column(String(256))
```
Comment (iamvishnuks) [inline model/discover.py:16]: project and hostname should be composite primary keys

File: model/disk.py [hunk 5]
```diff
@@ -1,25 +1,19 @@
...
-    mnt_path = StringField(required=True, max_length=150)
-    disk_id = StringField(required=True, max_length=550)
-    meta = {
-        'indexes': [
-            {'fields': ('host', 'project', 'mnt_path'), 'unique': True}
-        ]
-    }
\ No newline at end of file
+    id = Column(String(40), primary_key=True)
+    created_at = Column(DateTime(timezone=True), nullable=False)
```
Comment (iamvishnuks) [inline model/disk.py:10]: project relationship is missing and host, project and mnt_path should be composite primary key

File: model/machines.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    
+    __tablename__ = 'vm'
+
+    id = Column(String(40), primary_key=True)
+    created_at = Column(DateTime(timezone=True), nullable=False)
+    updated_at = Column(DateTime(timezone=True), nullable=False)
+    is_deleted = Column(Boolean, nullable=False, default=False)
+    is_locked = Column(Boolean, nullable=False, default=False)
+    is_enabled = Column(Boolean, nullable=False, default=True)
+    blueprint = Column(String(40), ForeignKey("blueprint.id"), nullable=False)
```
Comment (iamvishnuks) [inline model/machines.py:15]: blueprint and hostname should be composite primary key

File: model/network.py [hunk 7]
```diff
@@ -1,44 +1,35 @@
...
+    created_at = Column(DateTime(timezone=True), nullable=False)
+    updated_at = Column(DateTime(timezone=True), nullable=False)
+    is_deleted = Column(Boolean, nullable=False, default=False)
+    blueprint = Column(String(40), ForeignKey("blueprint.id"), nullable=False)
+    name = Column(String(256), primary_key=True)
+    cidr = Column(String(256))
+    created = Column(Boolean, nullable=False, default=False)
+    target_network_id = Column(String(256))
+    ig_id = Column(String(256))
+    route_table = Column(String(256))
```
Comment (iamvishnuks) [inline model/network.py:20]: cidr, project and nw_name should be composite primary keys

File: model/nodes.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    __tablename__ = 'node'
+
+    id = Column(String(40), primary_key=True)
+    created_at = Column(DateTime(timezone=True), nullable=False)
+    updated_at = Column(DateTime(timezone=True), nullable=False)
+    is_deleted = Column(Boolean, nullable=False, default=False)
+    hosts = Column(String(5120), nullable=False)
+    username = Column(String(256), nullable=False)
+    password = Column(String(256), nullable=False)
+    project = Column(String(40), ForeignKey("project.id"), nullable=False)
```
Comment (iamvishnuks) [inline model/nodes.py:16]: project and hostname should be composite primary key

**Review by iamvishnuks — APPROVED** [review]

**Pull request merged**